java-checks-test-sources/src/main/java/checks/regex/VerboseRegexCheck.java

```diff
@@ -0,0 +1,58 @@
+package checks.regex;
+
+import java.util.regex.Pattern;
+
+public class VerboseRegexCheck {
+  void nonCompliant() {
+    Pattern.compile("[0-9]"); // Noncompliant {{Use concise character class syntax '\\d' instead of '[0-9]'.}}
+    Pattern.compile("[^0-9]"); // Noncompliant  {{Use concise character class syntax '\\D' instead of '[^0-9]'.}}
+    Pattern.compile("something[0-9]somethingElse"); // Noncompliant [[sc=31;ec=36]] {{Use concise character class syntax '\\d' instead of '[0-9]'.}}
+
+    Pattern.compile("[A-Za-z0-9_]");    // Noncompliant  {{Use concise character class syntax '\\w' instead of '[A-Za-z0-9_]'.}}
+    Pattern.compile("[0-9_A-Za-z]");    // Noncompliant
+    Pattern.compile("[^A-Za-z0-9_]");   // Noncompliant  {{Use concise character class syntax '\\W' instead of '[^A-Za-z0-9_]'.}}
+    Pattern.compile("[^0-9_A-Za-z]");   // Noncompliant
+
+    Pattern.compile("x{0,1}");          // Noncompliant {{Use concise quantifier syntax '?' instead of '{0,1}'.}}
+    Pattern.compile("x{0,1}?");         // Noncompliant {{Use concise quantifier syntax '?' instead of '{0,1}'.}}
+    Pattern.compile("x{0,}");           // Noncompliant {{Use concise quantifier syntax '*' instead of '{0,}'.}}
+    Pattern.compile("x{0,}?");          // Noncompliant {{Use concise quantifier syntax '*' instead of '{0,}'.}}
+    Pattern.compile("x{1,}");           // Noncompliant {{Use concise quantifier syntax '+' instead of '{1,}'.}}
+    Pattern.compile("x{1,}?");          // Noncompliant {{Use concise quantifier syntax '+' instead of '{1,}'.}}
+    Pattern.compile("x{2,2}");          // Noncompliant {{Use concise quantifier syntax '{2}' instead of '{2,2}'.}}
+    Pattern.compile("x{2,2}?");         // Noncompliant {{Use concise quantifier syntax '{2}' instead of '{2,2}'.}}
+
+    Pattern.compile("[\\W\\w]"); // FN: {{Use concise character class syntax '.' instead of '[\\w\\W]'.}}
+    Pattern.compile("[\\d\\D]"); // FN: {{Use concise character class syntax '.' instead of '[\\d\\D]'.}}
+    Pattern.compile("[\\s\\S]", Pattern.DOTALL); // FN: {{Use concise character class syntax '.' instead of '[\\s\\S]'.}}
+    Pattern.compile("(?s)[\\s\\S]"); // FN: {{Use concise character class syntax '.' instead of '[\\s\\S]'.}}
```

Comment:
Should be fixed by https://github.com/SonarSource/sonar-analyzer-commons/pull/218.

Reply:
Merged on commons!